lib/oplog/tail.go

```diff
@@ -366,16 +360,77 @@ func closeCursor(cursor *mongo.Cursor) {
 	}
 }
 
-// unmarshalEntry unmarshals a single entry from the oplog.
+// unmarshalEntryMetadata processes the top-level data from an entry and returns a rawOplogEntry object.
```

Comment:
It might be nice to say a bit more about the performance issue with the usual unmarshalling, explaining why we need to do all these manual lookups.